services/s3/src/main/java/software/amazon/awssdk/services/s3/handlers/ErrorOn200Interceptor.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.s3.handlers;
+
+import static software.amazon.awssdk.utils.FunctionalUtils.invokeSafely;
+
+import java.io.BufferedInputStream;
+import java.util.Arrays;
+import java.util.List;
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.events.StartElement;
+import javax.xml.stream.events.XMLEvent;
+import software.amazon.awssdk.http.SdkHttpFullResponse;
+import software.amazon.awssdk.interceptor.Context;
+import software.amazon.awssdk.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.interceptor.ExecutionInterceptor;
+import software.amazon.awssdk.services.s3.model.CopyObjectRequest;
+
+public class ErrorOn200Interceptor implements ExecutionInterceptor {
+
+    private static final String ERROR_TAG = "Error";
+
+    private static final List<Class> OPERATIONS_WITH_200_ERROR = Arrays.asList(CopyObjectRequest.class);
+
+    private static final XMLInputFactory XML_INPUT_FACTORY = XMLInputFactory.newInstance();
+
+    public SdkHttpFullResponse modifyHttpResponse(Context.ModifyHttpResponse context,
```

Comment:
I'm 99.99% sure this won't work for async. I believe we buffer data after we call this method so content will be null. That said it might be possible to get it to work with async if we move where we call this method (might mean duplicate code paths for streaming vs non-streaming, not sure on that). Alternatively we can do it in the response handler similar to how you did it in V1 I believe.

Reply:
@spfink Can you respond to Andrew's comment

Reply:
We'd talked about this offline. I'll go ahead and close this. I've got a new response handler that will work for this case, just need to wire it up. Will open a new PR this week.